File: haskell/clockhands.hs
import Text.Printf (printf)

antiParallel :: Int -> (Int,Int,Int)
antiParallel n = (if h == 0 then 12 else h, m, s)
    where t = truncate $ (fromIntegral n + 0.5) * 43200 / 11
          (h, ms) = t `divMod` 3600
          (m, s) = ms `divMod` 60

main = mapM_ (putStrLn . renderTime . antiParallel) [0..10]
    where renderTime (h,m,s) = printf "%02d:%02d:%02d" h m s

File: ruby/clockhands.rb
0.upto 10 do |i|
  t = (i + 0.5) * 43200 / 11
  h, t = t.divmod 3600
  m, s = t.divmod 60
  printf("%02d:%02d:%02d\n", (h == 0 ? 12 : h), m, s)
end